src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -1396,6 +1396,60 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
         GenTree* op3 = nullptr;
         GenTree* op4 = nullptr;
 
+        switch (numArgs)
+        {
+            case 4:
+                op4 = getArgForHWIntrinsic(sigReader.GetOp4Type(), sigReader.op4ClsHnd);
+                op4 = addRangeCheckIfNeeded(intrinsic, op4, mustExpand, immLowerBound, immUpperBound);
+                op3 = getArgForHWIntrinsic(sigReader.GetOp3Type(), sigReader.op3ClsHnd);
+                op2 = getArgForHWIntrinsic(sigReader.GetOp2Type(), sigReader.op2ClsHnd);
+                op1 = getArgForHWIntrinsic(sigReader.GetOp1Type(), sigReader.op1ClsHnd);
+                break;
+
+            case 3:
+                op3 = getArgForHWIntrinsic(sigReader.GetOp3Type(), sigReader.op3ClsHnd);
```

Comment:
Is there no case where a range check is needed for 3 arguments?

Should there be an assert to validate that?

Reply:
> Is there no case where a range check is needed for 3 arguments?

That seems to be the case.

 The case for 3 args is checked further down and depending on the intrinsics, does the `addRangeCheckIfNeeded` on appropriate arg. Not sure if we should still add assert.